Source/Service/CotterAPIService.swift

```diff
@@ -216,6 +218,76 @@ public class CotterAPIService {
         )
     }
     
+    public func getBiometricStatus(
+        cb: HTTPCallback
+    ) {
+        let internalCb = CotterCallback()
+        
+        guard let pubKey = KeyGen.pubKey else {
+            internalCb.internalErrorHandler(err: "Unable to attain user's public key!")
+            return
+        }
+        
+        let pubKeyBase64 = CryptoUtil.keyToBase64(pubKey: pubKey)
+        
+        let method = "GET"
+        let path = "/user/enrolled/" + CotterAPIService.shared.userID! + "/BIOMETRIC/" + pubKeyBase64
+        
+        self.http(
+            method: method,
+            path: path,
+            body: nil,
+            cb: cb
+        )
+    }
+    
+    public func updateBiometricStatus(
+        enrollBiometric: Bool,
+        cb: HTTPCallback
+    ) {
+        let internalCb = CotterCallback()
+        
+        guard let pubKey = KeyGen.pubKey else {
+            internalCb.internalErrorHandler(err: "Unable to attain user's public key!")
+            return
+        }
+        
+        let pubKeyBase64 = CryptoUtil.keyToBase64(pubKey: pubKey)
+        
+        let method = "PUT"
+        let path = "/user/" + CotterAPIService.shared.userID!
+        let data: [String: Any] = [
+            "method": "BIOMETRIC",
+            "enrolled": enrollBiometric,
+            "code": pubKeyBase64
+        ]
+        
+        let body = try? JSONSerialization.data(withJSONObject: data)
+        
+        // If we are trying to disable biometrics, we need
+        // to delete the public/private key pair as well after successul HTTP Response
+        if !enrollBiometric {
+            func internalSuccessCb() {
+                do {
+                    try KeyGen.clearKeys()
+                } catch let err {
+                    internalCb.internalErrorHandler(err: err.localizedDescription)
+                    return
+                }
+            }
+            
+            internalCb.internalSuccessFunc = internalSuccessCb
```

Comment:
I see what you're trying to do, I might have a better idea,

```
var oldHandler = cb.successfulFunc
cb.successfulFunc = { (data: Data?) in 
  oldHandler(data:data)
  internalSuccessCb()
}
```

this way, you don't need internal callback protocol.

Reply:
nvm, it doesn't work, `successFunc` is inaccessible from cb.

may need to find other solution. but it's not priority now.